### ISSUE TITLE
Contradictory comment in atoms/atom_types.script 

### ISSUE THREAD

**Issue by kasimebrahim** [opencog/atomspace#2400]
In atom_types.script there is a comment about the specification of GLOB_NODE that says 

> the globbed values are meant to be of all same type.

On the other hand if I execute the below 
<pre>
(List (Concept "A") (Predicate "P"))
(cog-execute! (Get (List (Glob "$X"))))
</pre>
the result would be
<pre>
(SetLink
    (ListLink
        (ConceptNode "A")
        (PredicateNode "P")))
</pre>

which is contradictory with the above comment.

**Comment by ngeiswei** [comment]
Thanks @kasimebrahim. I suggest to remove the comment (and maybe point to the wiki), unless @linas has another suggestion.

**Comment by linas** [comment]
I guess the comment is misleading.  However, something like this should work:
```
(Get
    (TypedVariable (Glob "x") (Type 'ConceptNode))
    (List (Glob "x")))
```

and also this
```
(Get
    (TypedVariable (Glob "x")
           (TypeChoice (Type 'ConceptNode)) (Type 'PredicateNode)))
    (List (Glob "x")))
```
I doubt that either have unit tests, and would not be surprised if the type checks fail. .. but maybe it "just works" already? In which case, unit tests are still useful to have.

**Comment by linas** [comment]
I just tried it, they all work fine.
```
(List (Concept "B") (Concept "C"))
(cog-execute! (Get (TypedVariable (Glob "x") (Type 'ConceptNode)) (List (Glob "x"))))
(cog-execute! (Get (List (Glob "$X"))))
(cog-execute! (Get (TypedVariable (Glob "x") (TypeChoice (Type 'ConceptNode) (Type 'PredicateNode))) (List (Glob "x"))))
```
all return as expected.

**Comment by linas** [comment]
fixed in #2407
